File: Signiture-Backend/kotlin/study/src/main/java/default_value/Exam.java
package default_value;

import java.time.LocalDateTime;
import java.time.format.DateTimeFormatter;
import java.util.Optional;

public class Exam {

    public Exam(Store store) {
        var storeRegisterAt = toLocalDateTimeString(store.getRegisterAt());
        System.out.println(storeRegisterAt);
    }

    private String toLocalDateTimeString(LocalDateTime localDateTime) {
        LocalDateTime temp = Optional.ofNullable(localDateTime).orElse(LocalDateTime.now());
        return temp.format(DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss"));
    }

    public static void main(String[] args) {
        var store = new Store();
        new Exam(store);
    }

    static class Store {

        private LocalDateTime registerAt;

        public LocalDateTime getRegisterAt() {
            return registerAt;
        }

        public void setRegisterAt(LocalDateTime registerAt) {
            this.registerAt = registerAt;
        }
    }
}

File: Signiture-Backend/kotlin/study/src/main/kotlin/default_value/Exam.kt
package pcy.study.value

import java.time.LocalDateTime
import java.time.format.DateTimeFormatter

fun main() {
    Exam(Store())
    println(DateTimeUtil().localDateTimeToString())
}

class Exam(store: Store) {

    // 초기화 시에 호출되는 Scope
    init {
        val storeRegisterAt = toLocalDateTimeString(store.registerAt)
        println(storeRegisterAt)
    }

    private fun toLocalDateTimeString(localDateTime: LocalDateTime?): String {
        val temp = localDateTime ?: LocalDateTime.now()
        return temp.format(DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss"))
    }
}

data class Store(
    var registerAt: LocalDateTime? = null
)